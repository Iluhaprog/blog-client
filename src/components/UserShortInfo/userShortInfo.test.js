import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { UserShortInfo } from '.';
import configureMockStore from 'redux-mock-store'
const mockStore = configureMockStore();

test('Test user short info view', () => {
    const store = mockStore({
        user: {
            firstName: 'Test firstname',
            lastName: 'Test lastname'
        }
    });
    
    const { asFragment } = render(
        <Provider store={store}>
            <UserShortInfo />
        </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
});


test('Test user short info view (row)', () => {
    const store = mockStore({
        user: {
            firstName: 'Test firstname',
            lastName: 'Test lastname'
        }
    });
    
    const { asFragment } = render(
        <Provider store={store}>
            <UserShortInfo row={true} />
        </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
});

test('Test user short info view (width, height)', () => {
    const store = mockStore({
        user: {
            firstName: 'Test firstname',
            lastName: 'Test lastname'
        }
    });
    
    const { asFragment } = render(
        <Provider store={store}>
            <UserShortInfo row={true} width={30} height={30} />
        </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
});