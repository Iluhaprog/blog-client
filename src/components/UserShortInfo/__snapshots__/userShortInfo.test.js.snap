// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test user short info view 1`] = `
<DocumentFragment>
  <div
    class="user-short-info column column_ai-c"
  >
    <div
      class="round-box"
    >
      <img
        alt="Test firstname-Test lastname"
        height="65"
        src="default.png"
        width="65"
      />
    </div>
    <p>
      Test firstname Test lastname
    </p>
  </div>
</DocumentFragment>
`;
